pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/ddp/kitrequest/KitRequestDao.java

```diff
@@ -35,17 +40,20 @@ public class KitRequestDao implements Dao<KitRequestDto> {
 
     public static final String BY_INSTANCE_ID = " WHERE kr.ddp_instance_id = ?";
 
-    public static final String SQL_GET_KIT_REQUEST_ID = "SELECT  ddp_kit_request_id  FROM  ddp_kit_request";
-
-    public static final String BY_BSP_COLLABORATOR_SAMPLE_ID = " WHERE bsp_collaborator_sample_id = ?";
-
     public static final String SQL_GET_KIT_REQUEST =
             "SELECT  req.ddp_kit_request_id,   req.ddp_instance_id,  req.ddp_kit_request_id,  req.kit_type_id,"
                     + "req.bsp_collaborator_participant_id,  req.bsp_collaborator_sample_id,  req.ddp_participant_id, "
                     + "req.ddp_label,  req.created_by,  req.created_date,  req.external_order_number, "
                     + "req.external_order_date,  req.external_order_status,  req.external_response,  req.upload_reason, "
                     + "req.order_transmitted_at,  req.dsm_kit_request_id  FROM  ddp_kit_request req";
 
+    public static final String SQL_GET_SAMPLE_BY_BSP_COLLABORATOR_SAMPLE_ID = " SELECT * FROM  ddp_kit_request r LEFT JOIN ddp_kit k "
+            + " ON (r.dsm_kit_request_id = k.dsm_kit_request_id) WHERE bsp_collaborator_sample_id = ?";
+    public static final String SQL_RESAMPLE_KIT = " UPDATE ddp_kit_request SET bsp_collaborator_sample_id = ?, ddp_participant_id = ?, "
+            + "bsp_collaborator_participant_id = ? WHERE dsm_kit_request_id in  "
+            + "(SELECT dsm_kit_request_id FROM (SELECT * FROM ddp_kit_request) as tbl WHERE bsp_collaborator_sample_id = ?) "
```

Comment:
nit: if all we need is a `dsm_kit_request_id`, then do `(SELECT dsm_kit_request_id FROM ddp_kit_request)` instead of `select *`.